### PR TITLE
global: renamed cpu-conf.h into cpu_conf.h

### PR DESCRIPTION
**Pull request by haukepetersen**
This was annoying me for some time: all our other headers in that context use underscores...

You can verify this with searching from the RIOT root for the old `cpu-conf.h` in the code and the file tree with:
```
git grep cpu-conf.h
find . -name 'cpu-conf.h'
```
and there should be no occurrences left.

Further a passing Travis should show that everything builds fine.

### CODE REVIEW

**Comment by OlegHahm** [comment]
I can predict that it will take me a while to get used to this.

**Comment by haukepetersen** [comment]
but do you object in general or is it just the 'getting' used to it?

**Comment by OlegHahm** [comment]
No, I don't object. Unifying in general is a good thing. There are a couple of more places in RIOT where spaces and underscores are mixed quite arbitrary (e.g. iot-lab_M3...).

**Comment by gebart** [comment]
Might be time to look over the documentation for those CPU configurations. This is not the scope of the PR though, so I vote for ignoring Travis in this particular case. The errors are because Travis thinks the renamed files are new.
 
```
WARNING: The following modified files generate doxygen warnings:
boards/pttu/include/board.h
cpu/cortex-m0_common/include/cpu.h
cpu/cortex-m3_common/include/cpu.h
cpu/cortex-m4_common/include/cpu.h
cpu/msp430-common/include/cpu.h
cpu/x86/include/cpu.h
sys/include/net/ng_pktbuf.h
ERROR: The following new files generate doxygen warnings:
cpu/lpc2387/include/cpu_conf.h
cpu/mc1322x/include/cpu_conf.h
cpu/native/include/cpu_conf.h
cpu/samd21/include/cpu_conf.h
cpu/stm32f1/include/cpu_conf.h
```

**Comment by haukepetersen** [comment]
Agree with @gebart. I will prepare another PR to fix those doxygen issues. So anyone against pressing merge on this one?

**Comment by OlegHahm** [comment]
Can we convince Travis to run the missing tests somehow?

**Comment by haukepetersen** [comment]
I will just fix the doxygen issues in this PR.

**Comment by haukepetersen** [comment]
Those doxygen errors should be fixed now, lets see what Travis says about this...

**Comment by gebart** [comment]
Building application: driver_nrfmin (23/83) ....
		failed: pca10000, pca10005, yunjia-nrf51822
		failed .. retrying: airfy-beacon

**Comment by haukepetersen** [comment]
rebased. Now Travis should run fine.

**Comment by haukepetersen** [comment]
rebased again.

**Comment by haukepetersen** [comment]
and again. Travis seemed to fail to build for the `samr21-xpro`, but I could not reproduce this locally. Let's see if the latest rebase fixed this.

**Comment by haukepetersen** [comment]
rebased and fixed issue with the `saml21-xpro`, now Travis should be happy. Anyone to ack this?

File: boards/pttu/include/board.h [hunk 1]
```diff
@@ -24,23 +24,40 @@
...
 
 #include "lpc2387.h"
-#include "cpu-conf.h"
+#include "cpu_conf.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+/**
```
Comment (kaspar030) [inline boards/pttu/include/board.h]: Can you remove these from the PR?

Comment (kaspar030) [inline boards/pttu/include/board.h]: nevermind.

**Comment by kaspar030** [comment]
ACK when travis is happy.

**Comment by haukepetersen** [comment]
so just about 2 more days until merging :-)

**Comment by kaspar030** [comment]
I started my own compile-test, so maybe 90 minutes...

**Comment by kaspar030** [comment]
local test ran fine, static-tests fine, ACK & GO!

**Comment by haukepetersen** [comment]
nice!

**Pull request merged**